src/main/java/yevhenii/lostfilmdemo/repository/impl/JooqTVSeriesRepository.java

```diff
@@ -0,0 +1,133 @@
+package yevhenii.lostfilmdemo.repository.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.jooq.DSLContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+import yevhenii.lostfilmdemo.entity.TVSeries;
+import yevhenii.lostfilmdemo.jooq.generated.tables.records.TvSeriesRecord;
+import yevhenii.lostfilmdemo.repository.SchedulerRequestRepository;
+import yevhenii.lostfilmdemo.repository.UserRequestRepository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static yevhenii.lostfilmdemo.jooq.generated.tables.TvSeries.TV_SERIES;
+
+@Repository
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class JooqTVSeriesRepository implements SchedulerRequestRepository, UserRequestRepository {
+    private final DSLContext jooq;
+//created
+    @Override
+    public TVSeries create(TVSeries tvSeries) {
+
+        TvSeriesRecord record = jooq.insertInto(TV_SERIES)
+                .set(createRecord(tvSeries))
+                .returning()
+                .fetchOne();
+        assert record != null;
+        return convertQueryResultToModelObject(record);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public TVSeries findByLink(String link) {
+
+        TvSeriesRecord record = jooq.selectFrom(TV_SERIES)
+                .where(TV_SERIES.LINK.equal(link))
+                .fetchOne();
+        if (record == null) throw new IllegalArgumentException("TV series not found by link " + link);
```

Comment:
the fact that there is no episode by link in DB is normal use case, return optional instead of exception